packages/components/app/styles/components/code-block/index.scss

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// CODE-BLOCK
+//
+
+
+// Note: "theme" contains just color variables and syntax highlighting styles
+@import "theme";
+
+// DIMENSIONS
+$code-block-line-numbers-width: 49px; // 3em = 49px
+$code-block-code-padding: 16px;
+
+// CODE-BLOCK PARENT/WRAPPER
+
+.hds-code-block {
+  position: relative;
+  overflow: hidden; // So header corners don't poke out from container
+  color: var(--code-block-color-foreground-primary);
+  font-family: var(--hds-typography-code-200);
+  background-color: var(--code-block-color-surface-primary);
+  // Note: may change border-radius to be toggled by an "@isRounded" property
+  border-radius: 6px;
+
+  pre,
+  code {
+    line-height: var(--token-typography-body-200-line-height); // = 1.4286 | 20px
+    white-space: pre;
+    text-align: left;
+    text-shadow: none;
+    word-wrap: normal;
+    word-break: normal;
+    word-spacing: normal;
+    tab-size: 4;
+    hyphens: none;
+
+    @media print { text-shadow: none; }
+  }
+}
+
+// CHILD COMPONENTS / ELEMENTS
+
+// Header (contains title & description)
+.hds-code-block__header {
+  display: flex;
+  flex-flow: column;
+  gap: 4px;
+  padding: 8px $code-block-code-padding;
+  background-color: var(--code-block-color-surface-faint);
+  border-bottom: 1px solid var(--code-block-color-border-strong);
+}
+
+.hds-code-block__description { color: var(--code-block-color-foreground-faint); }
+
+// Code
+.code-block__code {
+  display: block;
+  margin: 0;
+  padding: $code-block-code-padding;
+  overflow: auto;
+}
+
+// CopyButton
+.hds-code-block__copy-button {
+  // Overriding default colors
+  color: var(--code-block-color-foreground-primary);
+  background-color: var(--code-block-color-surface-faint);
+  border: 1px solid var(--code-block-color-border-strong);
+
+  &:hover,
+  &.mock-hover {
+    background-color: var(--code-block-color-surface-primary);
+    border-color: var(--code-block-color-border-strong);
+  }
+
+  &:active,
+  &.mock-active {
+    background-color: #2b303c; // TODO: can't find color defined in Figma
```

Comment:
The value is right here, I've inherited the foundation naming convention but with the locally scoped token this should likely be something like:

```
--code-block-color-surface-interactive-active: #2B303C;
background-color: var(--code-block-color-surface-interactive-active);
```

Reply:
Ok looks like it's very one-off though? I don't seen any other "surface-interactive-blah" variable names?

Reply:
(I'm updating this in the code for now anyway, thanks.)